### ISSUE TITLE
wired output of wh 

### ISSUE THREAD

**Issue by LightToYang** [ZJULearning/ttfnet#15]
the wl of  wh ouput are always 0, they are negative before relu

tensor([[[  0.0000, 116.5211,  42.4937, 122.5416],
         [  0.0000, 107.6900,  41.2038, 117.2426],
         [  0.0000, 125.4157,  44.4884, 128.1640],
         [  0.0000, 118.4363,  38.8104, 120.8187],
         [  0.0000, 144.1779,  51.2308, 147.0248],
         [  0.0000, 123.4721,  42.6045, 126.1252],
         [  0.0000, 115.3537,  38.5883, 124.3103],
         [  0.0000,  95.5488,  33.9789, 121.5930],
         [  0.0000, 137.6849,  48.2651, 129.0714],
         [  0.0000, 153.8599,  44.6644, 129.4741],
         [  0.0000,  79.8924,  34.0436, 128.6441],
         [  0.0000,  43.4521,  19.8980,  50.6122],
         [  0.0000, 114.4678,  53.5928, 117.2749],
         [  0.0000,  50.5073,  21.2092,  52.7150],
         [  0.0000,  37.3370,  20.9332,  60.4037],
         [  0.0000,  58.5186,  24.4326,  60.5577],
         [  0.0000,  56.8286,  23.6414,  60.3964],
         [  0.0000,  20.9488,  12.8595,  23.5858],
         [  0.0000,  20.5648,  14.1972,  24.3704],
         [  0.0000,  21.1813,  14.0686,  24.6013],
         [  0.0000,  25.0874,  15.2591,  25.0824],
         [  0.0000,  58.8945,   7.9307,  51.6445],
         [  0.0000,  20.0220,  14.1485,  22.9746],
         [  0.0000,  23.1689,  14.9487,  26.1784],
         [  0.0000, 105.1940,  18.1960, 107.0099],
         [  0.0000,  93.0207,  22.4406,  81.9989],
         [  0.0000, 161.1775,  69.2532, 120.0340],
         [  0.0000, 152.0186,  59.0491, 113.5726],
         [  0.0000,  30.4867,  19.4364,  45.0907],
         [  0.0000,  35.7395,  70.6150,  40.3779],
         [  0.0000,  19.9065,  13.7627,  22.8954],
         [  0.0000,  87.0246,  91.8417, 110.4563],
         [  0.0000,  47.9610,  25.7173,  49.1239],
         [  0.0000, 124.6373,  26.2414, 132.3073],
         [  0.0000,  42.9990,  52.1530,  58.9938],
         [  0.0000, 109.3388,  92.1436, 129.7543],
         [  0.0000,  43.4066,  26.2116,  48.3573],
         [  0.0000,  19.8315,  13.0908,  21.9701],
         [  0.0000, 136.0730,  28.4794, 128.4158],
         [  0.0000,  19.9120,  13.3052,  21.5078],
         [  0.0000,  20.3097,  13.3365,  23.5736],
         [  0.0000,  20.0541,  13.0240,  23.1604],
         [  0.0000,  86.9304,  11.6221,  84.4194],
         [  0.0000, 108.4659,  18.4569,  93.0499],
         [  0.0000,  19.3067,  13.9523,  19.6273],
         [  0.0000, 123.7198,  23.3926, 114.5532],
         [  0.0000,  23.0683,  15.0457,  27.2681],
         [  0.0000,  11.1033,   5.4479,  18.0002],
         [  0.0000,  31.8376,  16.7024,  35.1585],
         [  0.0000,  65.6375,  32.1086,  78.4625],
         [  0.0000, 126.2547,  46.1084, 141.0742],
         [  0.0000,  49.8713,  16.5768,  61.0893],
         [  0.0000,  97.8429, 104.7201, 100.0325],
         [  0.0000,  59.1779,  41.5078, 135.6670],
         [  0.0000,  76.3698,  10.7484,  72.4081],
         [  0.0000,  74.9829,  46.3468, 155.4478],
         [  0.0000, 119.8739,  24.9125, 133.0377],
         [  0.0000,  10.6555,   4.1952,  17.6679],
         [  0.0000,  63.5594,  30.0634,  88.3806],
         [  0.0000,  88.6980,  93.6552,  72.6249],
         [  0.0000,  19.6395,  13.0398,  20.9386],
         [  0.0000, 128.0675,  22.5303, 117.6851],
         [  0.0000,  63.3495,  29.9718,  67.9388],
         [  0.0000,  18.8939,  13.5181,  18.6566],
         [  0.0000,   7.2600,   6.0671,  11.0599],
         [  0.0000,   7.3105,   6.0417,  11.0529],
         [  0.0000,  43.4574,  15.4914,  53.1781],
         [  0.0000,   7.0489,   5.8090,  10.0173],
         [  0.0000,   7.1726,   5.9593,  10.8656],
         [  0.0000,  62.8165,  34.6642,  66.0653],
         [  0.0000,  51.7567,  10.6115,  50.0044],
         [  0.0000,  92.7709, 105.0076, 102.7054],
         [  0.0000,  39.5032,  23.0723,  42.0505],
         [  0.0000,  47.5203,  42.7517, 108.0348],
         [  0.0000,   6.9966,   5.8431,  10.0747],
         [  0.0000, 113.8149, 107.7129, 119.9966],
         [  0.0000, 120.9095,  43.7954, 145.0118],
         [  0.0000,   7.0511,   5.7632,   9.8832],
         [  0.0000,  19.7299,  12.0109,  17.0321],
         [  0.0000,   7.3372,   5.9645,  10.9145],
         [  0.0000,  90.5423, 101.5567,  97.5997],
         [  0.0000,   7.0606,   5.7998,  10.0662],
         [  0.0000, 117.0862,  46.2676, 152.8364],
         [  0.0000, 144.9170,  44.8524,  92.6930],
         [  0.0000, 105.6485,  17.5557, 119.2687],
         [  0.0000, 105.2603, 105.4551, 105.4100],
         [  0.0000,   7.4934,   6.0036,  11.0666],
         [  0.0000,  78.2761, 101.2695,  83.9413],
         [  0.0000,  34.0802,  21.5975,  41.9197],
         [  0.0000,  37.6092,  22.0470,  35.8518],
         [  0.0000,  52.5578,  38.3820,  74.1862],
         [  0.0000,  79.0693,  61.6724, 153.6753],
         [  0.0000,  10.1691,   5.8722,  13.3266],
         [  0.0000,  73.8878,  53.4783, 138.8358],
         [  0.0000,   7.8271,   6.2989,  11.5233],
         [  0.0000,  62.8140,  23.9927,  66.1418],
         [  0.0000,   7.9598,   6.2933,  11.6692],
         [  0.0000, 111.2614, 103.2351, 134.6110],
         [  0.0000,  69.5796,  87.4113,  71.6074],
         [  0.0000,  54.6593,  26.5077,  60.6459]]], device='cuda:0')

this is the detection result
![test](https://user-images.githubusercontent.com/17929130/68561551-e99e1d80-0480-11ea-9e44-c8ecac0b248d.jpg)

**Comment by LightToYang** [comment]
It is a bug while useing nn.Relu(inplace=True)

**Comment by ZNorah** [comment]
Hello, I also met this problem.
May I ask if you have solved this bug?
I just removed the relu in wh_layer.

**Comment by Serenade-ajp** [comment]
@LightToYang @ZNorah 
hello, I also meet the similar problem,  May you guys share how you solved this ?

**Comment by ZNorah** [comment]
I removed the last relu.



发自我的iPhone


------------------ Original ------------------

**Comment by Serenade-ajp** [comment]
@ZNorah 
You mean the relu in this "wh = F.relu(self.wh(x)) * self.wh_offset_base" ?

**Comment by ZNorah** [comment]
yes



发自我的iPhone


------------------ Original ------------------

**Comment by Serenade-ajp** [comment]
@ZNorah Ok, thanks for your reply, I will have a try.

**Comment by Serenade-ajp** [comment]
@ZNorah 
Hi, I have tried to remove the relu in  "wh = F.relu(self.wh(x)) * self.wh_offset_base", but the GIOU loss diverged at the begin of the training, and later I found it may be caused by the unsuitable setting of learning rate.
